### ISSUE TITLE
Improve generate full stack app with jdl

### ISSUE THREAD

**Issue by antarus** [jhipster/generator-jhipster#8547]
##### **Overview of the issue**
The new generation feature of the full stack is cool but there are some bug / non-user friendly points.

- [x] 1. I first generate my stack with the `jhipster import-jdl bug.jdl` command. 
Next, i modify the jdl file, adding a `countryCode String` attribute to the Countries entity, or update an application.
I executes the command `jhipster import-jdl bug.jdl`.
Selecting an option in case of conflict is ultra complicated as the applications are generated in parallel. I do not necessarily want to use the `--force` option each time.

- [x] 2.add an option to update only the entities.which would be an equivalent of the option `--json-only'`and `for f in `ls .jhipster`; do jhipster entity $ {f%. *}; done`

- [x] 3. If I run the `jhipster import-jdl bug.jdl --json-only --ignore-application` command with or without making a change since the last command, , the generator is blocked after `Entity JSON files created . Entity generation skipped.`. obliged to do a CTRL + C.

- [x] 4. idem after modifying an entity the command `for f in `ls .jhipster`; do jhipster entity $ {f%. *} --force;`, the generator is blocked after `Congratulations, JHipster execution is complete!`

##### **Motivation for or Use Case**
Improve experience

##### **Reproduce the error**
`jhipster import-jdl bug.jdl`
updte an entity
`jhipster import-jdl bug.jdl`
try to choose an option for the conflict

##### **JHipster Version(s)**
5.4.2

##### **JHipster configuration**
<details><summary>jdl</summary>
application {
	config {
    	baseName gateway
        applicationType gateway
    }

    entities *
}
application {
	config {
    	baseName micro
        applicationType microservice
    }
    entities *
}

entity Region {
	regionName String
}

entity Country {
	countryName String
}

// an ignored comment
/** not an ignored comment */
entity Location {
	streetAddress String,
	postalCode String,
	city String,
	stateProvince String
}

entity Department {
	departmentName String required
}

/**
 * Task entity.
 * @author The JHipster team.
 */
entity Task {
	title String,
	description String
}

/**
 * The Employee entity.
 */
entity Employee {
	/**
	* The firstname attribute.
	*/
	firstName String,
	lastName String,
	email String,
	phoneNumber String,
	hireDate Instant,
	salary Long,
	commissionPct Long
}

entity Job {
	jobTitle String,
	minSalary Long,
	maxSalary Long
}

entity JobHistory {
	startDate Instant,
	endDate Instant,
	language Language
}

enum Language {
    FRENCH, ENGLISH, SPANISH
}

relationship OneToOne {
	Country{region} to Region
}

relationship OneToOne {
	Location{country} to Country
}

relationship OneToOne {
	Department{location} to Location
}

relationship ManyToMany {
	Job{task(title)} to Task{job}
}

// defining multiple OneToMany relationships with comments
relationship OneToMany {
	Employee{job} to Job,
	/**
	* A relationship
	*/
	Department{employee} to
	/**
	* Another side of the same relationship
	*/
	Employee
}

relationship ManyToOne {
	Employee{manager} to Employee
}

// defining multiple oneToOne relationships
relationship OneToOne {
	JobHistory{job} to Job,
	JobHistory{department} to Department,
	JobHistory{employee} to Employee
}

// Set pagination options
paginate JobHistory, Employee with infinite-scroll
paginate Job with pagination

// Use Data Transfert Objects (DTO)
// dto * with mapstruct

// Set service options to all except few
service all with serviceImpl except Employee, Job

// Set an angular suffix
// angularSuffix * with mySuffix

</details>

##### **Browsers and Operating System**
ubuntu 18.04

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
Thanks for your feedback @antarus

Another issue I saw is with `git commit` for the Gateway.
Fixing another issue for microservices (https://github.com/jhipster/generator-jhipster/pull/8545), we'll got the problem with microservices too.

**Comment by deepu105** [comment]
@antarus thanks for the detailed report. I'll look into it, in the meantime, if you have any improvement ideas please don't hesitate to PR

**Comment by deepu105** [comment]
adding a bounty on this since this is a tricky one

**Comment by antarus** [comment]
Another anomaly, on ubuntu, when the generation is over, the console seems 'broken'.
Nothing is written, forced to restart the terminal ...
test on different machine ubuntu 18.04 ..

**Comment by MathieuAA** [comment]
About your second point, have you thought of not importing the whole JDL file?

**Comment by MathieuAA** [comment]
I must not have understood the case then, my apologies

**Comment by deepu105** [comment]
@antarus need some clarifications about your points

2. Doesn't `--ignore-application` already do that? it creates only entities and ignores any app part
3. I don't understand you specify `--json-only` so it only creates the entity json files and skips any generation? what is the issue can you be more specific?
4. I don't understand this at all, can you explain clearly?

About your first issue, yes I need to work on that, but its gonna be tricky unless I disable parallel running.

**Comment by deepu105** [comment]
@antarus i'm still waiting for your input here

**Comment by antarus** [comment]
sorry I did not see and I was on vacation.

2. Yes, I found this solution after posting.
3. The generation of the code is skipped, but the application remains blocked after .. Obliged to make a CTRL + C to retrieve the prompt.
4. as the 3 but for entity generation. the entity is well generated but the application remains blocked after .. Obliged to make a CTRL + C to recover the prompt.

perhaps 3 and 4 are specific to ubuntu (test on several machine on ubuntu)

and for 1, yes 
apart disabled parallel running. there isn't much option.
perhaps, add a test at the beginning, if  files exist do not execute the code in parallel.
